flink-table/flink-sql-client/src/test/resources/sql/statement-set.q

```diff
@@ -0,0 +1,74 @@
+# statement-set.q - BEGIN STATEMENT SET, END
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to you under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# set default streaming mode and tableau result mode
+
+SET execution.runtime-mode = streaming;
+[INFO] Session property has been set.
+!info
+
+SET sql-client.execution.result-mode = tableau;
+[INFO] Session property has been set.
+!info
+
+create table src (
+  id int,
+  str string
+) with (
+  'connector' = 'values'
+);
+[INFO] Execute statement succeed.
+!info
+
+# ==========================================================================
+# test statement set
+# ==========================================================================
+
+BEGIN STATEMENT SET;
+[INFO] Begin the statement set.
+!info
+
+BEGIN STATEMENT SET;
+[ERROR] Could not execute SQL statement in statement set.
+!error
+
+create table test (
+id int,
+str string
+) with (
+'connector' = 'values'
+);
+[ERROR] Could not execute SQL statement in statement set.
+!error
+
+SELECT id, str FROM (VALUES (1, 'Hello World'), (2, 'Hi'), (2, 'Hi')) as T(id, str);
+[ERROR] Could not execute SQL statement in statement set.
+!error
+
+INSERT INTO src SELECT id, str
+FROM (VALUES (1, 'Hello World'), (2, 'Hi'), (2, 'Hi')) as T(id, str);
+[INFO] Add SQL update statement to the statement set.
+!info
+
+END;
+[INFO] Submitting SQL statement set to the cluster...
+[INFO] Table statement set has been successfully submitted to the cluster.
```

Comment:
I suggest to test this feature in dml-sync mode. Could you rebase the branch and take the dml-sync tests as an example?